src/pudl/output/ferc1.py

```diff
@@ -228,6 +228,55 @@ def _out_ferc1__yearly_plants_utilities(
     )
 
 
+## example/draft factory pattern
+specs = [
+    {
+        "name": "out_ferc1__yearly_purchased_power_and_exchanges_sched326",
+        "df1": "core_ferc1__yearly_purchased_power_and_exchanges_sched326",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+    {
+        "name": "out_ferc1__yearly_plant_in_service_sched204",
+        "df1": "core_ferc1__yearly_plant_in_service_sched204",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+    {
+        "name": "out_ferc1__yearly_balance_sheet_assets_sched110",
+        "df1": "core_ferc1__yearly_balance_sheet_assets_sched110",
+        "df2": "core_pudl__assn_ferc1_pudl_utilities",
+        "mg": "utility_id_ferc1",
+    },
+]
+
+
+# draft asset_factory
+def generate_asset_factory(spec):
+    @asset(io_manager_key="pudl_io_manager", compute_kind="Python")
+    def _asset() -> pd.DataFrame:
```

Comment:
The function which is used to define the asset needs to have arguments, because that's how Dagster infers the dependencies between assets. For more complex cases they can also be defined with arguments in the asset decorator, but I don't think that's necessary here. You can't just add them inside the inner function definition.

Search for "_asset_factory" in the code base to find other examples of how we're using this pattern.